corehq/apps/domain/urls.py

```diff
@@ -171,12 +172,16 @@
         name=AddCaseRepeaterView.urlname),
     url(r'^forwarding/new/(?P<repeater_type>\w+)/$', AddRepeaterView.as_view(), name=AddRepeaterView.urlname),
     url(r'^forwarding/test/$', test_repeater, name='test_repeater'),
+
     url(r'^forwarding/CaseRepeater/edit/(?P<repeater_id>\w+)/$', EditCaseRepeaterView.as_view(),
         {'repeater_type': 'CaseRepeater'}, name=EditCaseRepeaterView.urlname),
     url(r'^forwarding/FormRepeater/edit/(?P<repeater_id>\w+)/$', EditFormRepeaterView.as_view(),
         {'repeater_type': 'FormRepeater'}, name=EditFormRepeaterView.urlname),
-    url(r'^forwarding/edit/(?P<repeater_id>\w+)/$', EditRepeaterView.as_view(),
+    url(r'^forwarding/OpenmrsRepeater/edit/(?P<repeater_id>\w+)/$', EditOpenmrsRepeaterView.as_view(),
+        {'repeater_type': 'OpenmrsRepeater'}, name=EditOpenmrsRepeaterView.urlname),
+    url(r'^forwarding/(?P<repeater_type>\w+)/edit/(?P<repeater_id>\w+)/$', EditRepeaterView.as_view(),
```

Comment:
I am unable to test this since i blew up my local set up.

Earlier any repeater that was branched from `CaseRepeater` would still get `forwarding/CaseRepeater/edit/` matched since for the link `repeater.is_case_repeater` was `True` but this change makes all such repeaters to come through this route. Great if you have already checked that such repeaters are working fine and look the same when creating new and editing. I suspect it might miss some fields that are added in `CaseRepeater` by using `EditRepeaterView` instead of `CaseRepeaterView`.

Reply:
Do you mean all subclasses of CaseRepeater got the CaseRepeater form?

Did all subclasses of FormRepeater get the FormRepeater form? 

I've pushed another commit. The code feels wrong. And it's currently untested ... because Homeweek. (Labelling "Hold off".) But I'm curious to get your feedback regarding the approach. Would this, or a variation of this, result in the expected behaviour? Is it a reasonable approach? Is it the best approach?

Reply:
Dropping "Hold off". I've done a little rebasing to clean this up. Current functionality is preserved in that subclasses of `FormRepeater` use `EditFormRepeaterView`, subclasses of `CaseRepeater` use `EditCaseRepeaterView`, and other classes use `EditRepeaterView`. 

This change moves all routing to `urls.py`, so although it could be shorter, I think it improves separation of concerns, it's better than my first attempt, and it simplifies the template a little.